src/main/java/graphql/execution/incremental/DeferredCallContext.java

```diff
@@ -0,0 +1,40 @@
+package graphql.execution.incremental;
+
+import graphql.ExceptionWhileDataFetching;
+import graphql.GraphQLError;
+import graphql.Internal;
+import graphql.execution.ExecutionStrategyParameters;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Contains data relevant to the execution of a {@link DeferredCall}.
+ * <p>
+ * The responsibilities of this class are similar to {@link graphql.execution.ExecutionContext}, but restricted to the
+ * execution of a deferred call (instead of the whole GraphQL execution like {@link graphql.execution.ExecutionContext}).
+ * <p>
+ * Some behaviours, like error capturing, need to be scoped to a single {@link DeferredCall}, because each defer payload
+ * contains its own distinct list of errors.
+ */
+@Internal
+public class DeferredCallContext {
+
+    private final List<GraphQLError> errors = new CopyOnWriteArrayList<>();
+
+    public void onFetchingException(ExecutionStrategyParameters parameters, Throwable e) {
```

Comment:
rather than taking `ExecutionStrategyParameters` and hence tying this class absolutely to the ES classes - I would pass in the attrs you need

This is a minor tweak since this is an internal class

Reply:
yep, good one